Update PR template instructions in AGENTS.md (#170834)
home-assistant/core: .github/copilot-instructions.md

## Pull Requests

> - When opening a pull request, use the repository's PR template (`.github/PULL_REQUEST_TEMPLATE.md`). Do not remove any sections or checkboxes from the template.

Author: ```suggestion
- When opening a pull request, use the repository's PR template (`.github/PULL_REQUEST_TEMPLATE.md`). NEVER REMOVE ANYTHING from the template.
```
